src/tests/unittests/validation/RenderPipelineValidationTests.cpp

```diff
@@ -99,6 +114,72 @@ TEST_F(RenderPipelineValidationTest, CreationMissingProperty) {
     }
 }
 
+TEST_F(RenderPipelineValidationTest, BlendState) {
+    // Fails because blend state is set on a nonexistent color attachment
+    {
+        auto colorBuffer = device.CreateTextureBuilder()
+            .SetDimension(nxt::TextureDimension::e2D)
+            .SetExtent(640, 480, 1)
+            .SetFormat(nxt::TextureFormat::R8G8B8A8Unorm)
+            .SetMipLevels(1)
+            .SetAllowedUsage(nxt::TextureUsageBit::OutputAttachment)
+            .GetResult();
+        colorBuffer.FreezeUsage(nxt::TextureUsageBit::OutputAttachment);
+        auto colorView = colorBuffer.CreateTextureViewBuilder()
+            .GetResult();
+
+        auto renderpass = device.CreateRenderPassBuilder()
+            .SetAttachmentCount(2)
+            .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+            .AttachmentSetFormat(1, nxt::TextureFormat::R8G8B8A8Unorm)
+            .SetSubpassCount(1)
+            .SubpassSetColorAttachment(0, 0, 0)
+            .GetResult();
+
+        auto framebuffer = device.CreateFramebufferBuilder()
+            .SetRenderPass(renderpass)
+            .SetDimensions(640, 480)
+            .SetAttachment(0, colorView)
```

Comment:
nit: this should fail creation because we are missing one attachment. You don't need to add the validation in the patch, but it should SetAttachment(1) too (with a different texture too).